File: saxpy_loop.py
import time

import saxpy

x = [saxpy.XVAL] * saxpy.N
y = [saxpy.YVAL] * saxpy.N

print("N: {}".format(len(x)))

t0 = time.time()
for i in range(saxpy.N):
    y[i] += saxpy.AVAL * x[i]
t1 = time.time()

print("Elapsed: {} ms".format((t1 - t0) * 1000))
saxpy.verify(y)

File: src/saxpy_loop.jl
function saxpy( a, x, y )
    @simd for i=1:length(x)
        @inbounds y[i] += a*x[i]
    end
end

function flog( N )
    println("N: $N")
    
    const XVAL = Float32(10 * rand())
    const YVAL = Float32(10 * rand())
    const AVAL = Float32(10 * rand())
    
    x = zeros(Float32,N) + XVAL
    y = zeros(Float32,N) + YVAL    
    
    time = @elapsed saxpy(AVAL,x,y)
    
    time = time * 1000
    println("Elapsed: $time ms")
    
    err = 0.0
    for i in 1:N
        err += abs( y[i] - (YVAL + AVAL * x[i]) )
    end
    println("Error: $err")
end

flog(1 << 26)
